### PR TITLE
[WIP] PyTorch ResNet50

### PR DESCRIPTION
**Pull request by hanlint**
Adds a pytorch resnet-50 v1.5 implementation to the benchmark. The model is based on the `torchvision` ResNet-50v1.5 implementation. 

Roadmap for this PR to be ready to merge:

- [ ] Fix hparams and verify convergence (cc: @znado perhaps we can coordinate on the learning hyperparameters)
- [ ] Extend to multi-GPU
- [ ] Benchmark and optimize performance on 8x V100s (p3.16xlarge)
- [ ] Refactor boilerplate pytorch code shared across benchmarks

This is WIP, all comments welcome.

### CODE REVIEW

**Comment by hanlint** [comment]
@znado Running into an issue with the JAX rng. From a `python3` prompt:

```
import jax, os, struct

rng_seed = struct.unpack('q', os.urandom(8))[0]
rng = jax.random.PRNGKey(rng_seed)
```

Suddenly uses almost all of my GPU memory:
```
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 450.51.06    Driver Version: 450.51.06    CUDA Version: 11.3     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  Tesla V100-SXM2...  Off  | 00000000:00:04.0 Off |                    0 |
| N/A   37C    P0    58W / 300W |  14576MiB / 16160MiB |      0%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
```

Here's my jax versions:
```
jax==0.2.14
jaxlib==0.1.65+cuda110
```

**Comment by znado** [comment]
Ah yeah, jax is aggressive and reserves all GPU memory when imported. There are ways to disable this, but we recently agreed not to use jax for PyTorch RNGs so lets just skip the `rng = jax.random.PRNGKey(rng_seed)` part and use `rng_seed` directly in PyTorch. That way a PyTorch submitter doesn't need to install/import jax at all.

**Comment by znado** [comment]
Can you try pulling in [this PR](https://github.com/mlcommons/algorithmic-efficiency/pull/17) and seeing if it works for you? I tested it on both the PyTorch and Jax MNIST workloads and it worked, even when I didn't have Jax installed for the PyTorch impl.

File: workloads/resnet/resnet_pytorch/workload.py [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+
+  def preprocess_for_eval(
+      self,
+      raw_input_batch: spec.Tensor,
+      raw_label_batch: spec.Tensor,
+      train_mean: spec.Tensor,
+      train_stddev: spec.Tensor) -> spec.Tensor:
+    del train_mean
+    del train_stddev
+    return (raw_input_batch.float().to(DEVICE), raw_label_batch.to(DEVICE))
```
Comment (znado) [inline workloads/resnet/resnet_pytorch/workload.py:113]: we can delete these two functions. we can do the device transfer wherever you believe it makes the most sense, for the jax version that was inside [`build_input_queue(...)`](https://github.com/mlcommons/algorithmic-efficiency/pull/11/files#diff-d9615b1ae805f3c82746f0468d5d455431b83369e032299ef45334f7ba64cd46R239)

**Comment by runame** [comment]
Superseded by #45.

**Pull request closed without merging**